ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx

```diff
@@ -53,13 +56,16 @@ export const RecipientRow = ({ recipient }: { recipient?: Hex } = {}) => {
   const t = useI18nContext();
   const { currentConfirmation } = useConfirmContext<TransactionMeta>();
   const to = recipient ?? currentConfirmation?.txParams?.to;
+  const isBatchTransaction =
+    TransactionType.batch === currentConfirmation?.type;
 
-  if (!to || !isValidAddress(to)) {
+  if (!to || !isValidAddress(to) || isBatchTransaction) {
```

Comment:
Should we be future-proof and also check if `to` is equal to `from`?

Just in case we send batch transactions elsewhere in future.

Reply:
yes, I pushed a fix to check if `to` is equal to `from` as well.